configure.ac

```diff
@@ -640,6 +640,12 @@ AC_CHECK_FUNCS(explicit_bzero memset_explicit)
 AC_CHECK_FUNCS([ruserok_af ruserok], [break])
 AC_CHECK_FUNCS(close_range)
 
+dnl For module/pam_timestamp
+AC_CHECK_HEADERS([sys/random.h])
+AC_SEARCH_LIBS([getrandom], [bsd System], [
+  AC_CHECK_FUNCS(getrandom)
+])
```

Comment:
The reason why I suggested not to invest time into `AC_SEARCH_LIBS` approach is that it's too complicated to worth the effort.

The side effect of this `AC_SEARCH_LIBS` is that everything will be linked with the library that provides `getrandom`.
Since `getrandom` is only needed for pam_timestamp/hmac_openssl_wrapper.c, the linking with extra libraries should be limited to the linking of that file, that is, only when openssl is enabled and only when this file is being linked.  This implies that all these new checks are needed only for `use_openssl=yes` case and the extra libraries would have to be saved into a separate variable and used only in the linking command that needs it.

I still don't see why would you want to go into all this complexity in this project which doesn't seem to be used outside Linux ecosystem nowadays.

Reply:
Reason is mainly to be friendly to the person who may encounter compile errors on non-Linux platforms. ;-) I left a note re `libbsd`/`libSystem` in a comment; apart from that, it's a plain `AC_CHECK_FUNCS` call now.